Buildings/Fluid/CHPs/OrganicRankine/Cycle.mo

```diff
@@ -0,0 +1,493 @@
+    "Nominal pressure drop of the cold fluid in condenser"
+    annotation(Dialog(group="Condenser"));
+  parameter Modelica.Units.SI.TemperatureDifference dTPinCon(
+    final min = 0) = 10
+    "Condenser pinch point temperature difference"
+    annotation(Dialog(group="Condenser"));
+  parameter Boolean useLowCondenserPressureWarning = true
+    "If true, issues warning if pCon < 101325 Pa"
+    annotation(Dialog(group="Condenser"));
+  parameter Modelica.Units.SI.MassFlowRate mWor_flow_max(
+    final min = 0)
+    "Upper bound of working fluid flow rate"
+    annotation(Dialog(group="Cycle"));
+  parameter Modelica.Units.SI.MassFlowRate mWor_flow_min(
+    final min = 0)
+    "Lower bound of working fluid flow rate"
+    annotation(Dialog(group="Cycle"));
+  parameter Modelica.Units.SI.MassFlowRate mWor_flow_hysteresis
+    = mWor_flow_min + (mWor_flow_max - mWor_flow_min) * 0.1
+    "Hysteresis for turning off the cycle when flow too low"
+    annotation(Dialog(group="Cycle"));
+  parameter Modelica.Units.SI.Efficiency etaExp
+    "Expander efficiency"
+    annotation(Dialog(group="Cycle"));
+  parameter Modelica.Units.SI.Efficiency etaPum
+    "Pump efficiency"
+    annotation(Dialog(group="Cycle"));
+
+  Modelica.Blocks.Interfaces.RealOutput PEle(
+    final quantity="Power",
+    final unit="W") "Electrical power output from the expander"
+    annotation (Placement(
+        transformation(extent={{100,10},{140,50}}), iconTransformation(extent={{
+            70,20},{90,40}})));
+  Modelica.Blocks.Interfaces.BooleanInput ena
+    "Enable cycle; set false to force working fluid flow to zero" annotation (
+      Placement(transformation(extent={{-140,-20},{-100,20}}),
+        iconTransformation(extent={{-90,-10},{-70,10}})));
+  Modelica.Blocks.Interfaces.RealOutput QEva_flow(
+    final quantity="HeatFlowRate",
+    final unit="W") "Evaporator heat flow rate (positive)" annotation (
+      Placement(transformation(extent={{100,70},{140,110}}),iconTransformation(
+          extent={{-10,-10},{10,10}},
+        rotation=90,
+        origin={40,90})));
+  Modelica.Blocks.Interfaces.RealOutput QCon_flow(
+    final quantity="HeatFlowRate",
+    final unit="W") "Condenser heat flow rate (positive)" annotation (
+      Placement(transformation(extent={{100,-110},{140,-70}}),
+        iconTransformation(extent={{-10,-10},{10,10}},
+        rotation=-90,
+        origin={40,-90})));
+  Modelica.Blocks.Interfaces.BooleanOutput on_actual
+    "Actual on off status of the cycle" annotation (Placement(transformation(
+          extent={{100,-20},{140,20}}), iconTransformation(extent={{70,-10},{90,
+            10}})));
+  Modelica.Blocks.Interfaces.RealOutput PPum(
+    final quantity="Power",
+    final unit="W")
+    "Electrical power consumption of the pump" annotation (Placement(
+        transformation(extent={{100,-50},{140,-10}}),iconTransformation(extent={{70,-40},
+            {90,-20}})));
```

Comment:
Always use the convention that heat added the component is positive, and heat rejected is negative. (In a chiller, `QEva_flow` is a negative quantity.) 
Then, the instance `gai` can be removed.

Reply:
Addressed in fc78e7adf3feebc81b7de1e01b42e02001eb9172.
For the sake of the manuscript, do we make everything positive with some post processing? I know Modelica people would be accustomed to "power generated by the expander" being negative as per the sign convention. But the wider audience may find this odd. Also having everything on the positive side of the graph makes it much easier to see the scales, e.g. pump work is really small compared to expander output. This is the reason I opted not to follow the sign convention in developing this model.

Reply:
I would make the manuscript the same as the Modelica model, and we can always plot $-P_{ele}$.

Reply:
@hcasperfu : You still need to search the code for "(positive)" which are still in `Cycle.mo` and perhaps in other files.

Reply:
Removed those "(positive)" and "(negative)" in descriptions in 31253a4.

Reply:
In f56a791 I corrected the signs again. $\dot Q_{eva}$ should be positive and $\dot Q_{con}$ should be negative, from the cycle's stand point. This resulted both of them needing to be flipped by a gain block in the higher-level model `OrganicRankine.Cycle`.
Looking at the evaporator, the cycle takes heat (+) and the hot fluid loses heat (-); at the condenser, the cycle rejects heat (-) and the cold fluid absorbs heat (+).
Please let me know if I got this wrong.
I have also updated the manuscript.

Reply:
In 6977199, I changed signs according to our meeting.
$\dot Q_{eva}$ is negative and $\dot Q_{con}$ is positive. They follow the sign convention wrt the media (hot fluid in evaporator and cold fluid in condenser), not to the working fluid.
Power generation $P_{exp}$ is negative and power consumption $P_{pum}$ is positive.